index.bs

```diff
@@ -261,39 +259,78 @@ vrDisplay.requestPresent([{ source: canvas }]).then(function() {
 
 ## VRLayer ## {#interface-vrlayer}
 
-The {{VRLayer}} interface is provided to a {{VRDisplay}} and presented in the HMD.
+The {{VRLayer}} is the base interface for the various layer types that can be
+provided to a {{requestPresent()}} call to be presented in the HMD.
 
-<pre class="idl" id="vrlayer-dictionary">
+<pre class="idl" id="vrlayer-interface">
+interface VRLayer {};
+</pre>
+
+## CanvasVRLayer ## {#interface-canvasvrlayer}
+
+The {{CanvasVRLayer}} describes a layer containing head tracked stereo content
+rendered to a canvas with a WebGL context.
+
+<pre class="idl" id="canvasvrlayer-interface">
 typedef (HTMLCanvasElement or
-         OffscreenCanvas) VRSource;
+         OffscreenCanvas) CanvasVRSource;
 
-[Constructor(optional VRLayerInit layer)]
-interface VRLayer {
-  readonly attribute VRSource? source;
+[Constructor(optional CanvasVRSource source)]
+interface CanvasVRLayer : VRLayer {
```

Comment:
One area that I'm still waffling on is if we want to have two interfaces for each layer type: One for passing into `requestPresent` and one to be returned by `getLayers` (Which would likely be `CanvasVRLayerReadOnly` or something like that.) That way attributes that can't be set by the user (`renderWidth/Height`) are only in the ReadOnly variant and there's no confusion about what happens if you change the layer values after getting them out of `getLayers` like so:

```
var layers = vrDisplay.getLayers();
layers[0].setLeftBounds(l, b, r, t);
// Hey, WTF! Why didn't my layer bounds change like I asked?
```